tcp/client.go

```diff
@@ -0,0 +1,58 @@
+package tcp
+
+import (
+	"encoding/binary"
+	"net"
+
+	"github.com/influxdb/influxdb/data"
+)
+
+type Client struct {
+	conn net.Conn
+}
+
+func NewClient() *Client {
+	return &Client{}
+}
+
+func (c *Client) Dial(addr string) error {
+	conn, err := net.Dial("tcp", addr)
+	if err != nil {
+		return err
+	}
+
+	c.conn = conn
+	return nil
+}
+
+func (c *Client) WriteShardRequest(shardID uint64, points []data.Point) error {
+	mt := writeShardRequestMessage
+	if err := binary.Write(c.conn, binary.LittleEndian, &mt); err != nil {
```

Comment:
I was initially going to do that, but talked internally here and we felt that TLV (http://en.wikipedia.org/wiki/Type-length-value) was still the correct approach.  I don't have strong feelings either way.

I'm not sure how the proto buff is affected is I don't now which type of message I'm going to receive on the other end.  Maybe we should chat, I could be missing something.